### PR TITLE
Add livenessprobe to OneAgent from image

### PR DESCRIPTION
**Pull request by waodim**
## Description

Similar to the readinessProbe now the livenessProbe is set by checking depending if the Healthcheck is set in the dynakube status. Here, the livenesProbe will be set only if the readinessProbe is not nil.

## How can this be tested?

Deploy the operator, apply a dynakube where oneagents spawn and check their status, if the livenessProbe is set or not.

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/controllers/dynakube/oneagent/daemonset/daemonset.go [hunk 1]
```diff
@@ -213,18 +213,12 @@ func (dsInfo *builderInfo) podSpec() corev1.PodSpec {
...
-					Command: dsInfo.dynakube.Status.OneAgent.Healthcheck.Test,
-				},
-			},
-			InitialDelaySeconds: DefaultReadinessProbeInitialDelay,
-			PeriodSeconds:       int32(dsInfo.dynakube.Status.OneAgent.Healthcheck.Interval.Seconds()),
-			TimeoutSeconds:      int32(dsInfo.dynakube.Status.OneAgent.Healthcheck.Timeout.Seconds()),
-			FailureThreshold:    int32(dsInfo.dynakube.Status.OneAgent.Healthcheck.Retries),
-		}
+	if dsInfo.dynakube.NeedsOneAgentProbe() {
+		podSpec.Containers[0].ReadinessProbe = dsInfo.getProbe()
```
Comment (andriisoldatenko) [inline src/controllers/dynakube/oneagent/daemonset/daemonset.go:217]:
```suggestion
		podSpec.Containers[0]. LivenessProbe = dsInfo.getProbe()
```
Why not combine it?

File: src/controllers/dynakube/oneagent/daemonset/daemonset.go [hunk 2]
```diff
@@ -213,18 +213,12 @@ func (dsInfo *builderInfo) podSpec() corev1.PodSpec {
...
+	}
+
+	if podSpec.Containers[0].ReadinessProbe != nil {
```
Comment (andriisoldatenko) [inline src/controllers/dynakube/oneagent/daemonset/daemonset.go:220]:
```suggestion
```

File: src/controllers/dynakube/oneagent/daemonset/daemonset_test.go [hunk 3]
```diff
@@ -529,9 +529,49 @@ func TestPodSpecReadinessProbe(t *testing.T) {
 		assert.Equal(t, testCommands, podSpec.Containers[0].ReadinessProbe.Exec.Command)
 		assert.Equal(t, int32(interval.Seconds()), podSpec.Containers[0].ReadinessProbe.PeriodSeconds)
 		assert.Equal(t, int32(timeout.Seconds()), podSpec.Containers[0].ReadinessProbe.TimeoutSeconds)
-		assert.Equal(t, DefaultReadinessProbeInitialDelay, podSpec.Containers[0].ReadinessProbe.InitialDelaySeconds)
+		assert.Equal(t, DefaultProbeInitialDelay, podSpec.Containers[0].ReadinessProbe.InitialDelaySeconds)
 		assert.Equal(t, int32(retries), podSpec.Containers[0].ReadinessProbe.FailureThreshold)
 	})
+	t.Run("check livenessprobe if readiness probe is set", func(t *testing.T) {
```
Comment (andriisoldatenko) [inline src/controllers/dynakube/oneagent/daemonset/daemonset_test.go:535]:
```suggestion
	t.Run("check liveness probe if readiness probe is set", func(t *testing.T) {
```

File: src/controllers/dynakube/oneagent/daemonset/daemonset_test.go [hunk 4]
```diff
@@ -529,9 +529,49 @@ func TestPodSpecReadinessProbe(t *testing.T) {
...
+		podSpec := builder.podSpec()
+
+		assert.NotNil(t, podSpec.Containers[0].LivenessProbe)
+		assert.Equal(t, testCommands, podSpec.Containers[0].LivenessProbe.Exec.Command)
+		assert.Equal(t, int32(interval.Seconds()), podSpec.Containers[0].LivenessProbe.PeriodSeconds)
+		assert.Equal(t, int32(timeout.Seconds()), podSpec.Containers[0].LivenessProbe.TimeoutSeconds)
+		assert.Equal(t, DefaultProbeInitialDelay, podSpec.Containers[0].LivenessProbe.InitialDelaySeconds)
+		assert.Equal(t, int32(retries), podSpec.Containers[0].LivenessProbe.FailureThreshold)
+	})
+	t.Run("check livenessprobe is not set when readiness probe is not set", func(t *testing.T) {
```
Comment (andriisoldatenko) [inline src/controllers/dynakube/oneagent/daemonset/daemonset_test.go:560]:
```suggestion
	t.Run("check liveness probe is not set when readiness probe is not set", func(t *testing.T) {
```

File: src/controllers/dynakube/oneagent/daemonset/daemonset.go (hunk 1 above)
Comment (waodim) [reply src/controllers/dynakube/oneagent/daemonset/daemonset.go:217]: you are right, thanks. i got confused by the condition that liveness probe should only be set if readiness is set. :slightly_smiling_face:

Comment (waodim) [reply src/controllers/dynakube/oneagent/daemonset/daemonset.go:217]: fixed in [9863db2](https://github.com/Dynatrace/dynatrace-operator/pull/2115/commits/9863db2f7e301ed91d3595575008084c88a3ce09)

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**